runtime/src/pallet_assets_mock.rs

```diff
@@ -0,0 +1,244 @@
+// Mock implementation for Assets that satisfies token_gateway requirements
```

Comment:
`pallet_token_gateway` `Assets` config parameter requires only an implementer of `fungible::Mutate` and `fungible::metadata::Mutate` so you don't necessarily need to "replicate" the structure of a Pallet here

Reply:
true I removed a few things but kept the minimal needed for it to compile